tests/controls/test_explore.py

```diff
@@ -0,0 +1,267 @@
+import ipywidgets
+import pytest
+
+import seismometer.controls.explore as undertest
+
+
+class TestUpdatePlotWidget:
+    def test_init(self):
+        widget = undertest.UpdatePlotWidget()
+        assert widget.plot_button.description == undertest.UpdatePlotWidget.UPDATE_PLOTS
+        assert widget.plot_button.disabled is False
+        assert widget.code_checkbox.description == "show code"
+        assert widget.show_code is False
+
+    def test_plot_button_click(self):
+        count = 0
+
+        def on_click_callback(button):
+            nonlocal count
+            count += 1
+
+        widget = undertest.UpdatePlotWidget()
+        widget.on_click(on_click_callback)
+        widget.trigger()
+        assert count == 1
+
+    def test_disable(self):
+        widget = undertest.UpdatePlotWidget()
+        widget.disabled = True
+        assert widget.plot_button.disabled
+        assert widget.disabled
+
+    def test_enable(self):
+        widget = undertest.UpdatePlotWidget()
+        widget.disabled = True
+        widget.disabled = False
+        assert not widget.plot_button.disabled
+        assert not widget.disabled
+        assert widget.plot_button.description == undertest.UpdatePlotWidget.UPDATE_PLOTS
+
+    def test_toggle_code_checkbox(self):
+        count = 0
+
+        def on_toggle_callback(button):
+            nonlocal count
+            count += 1
+
+        widget = undertest.UpdatePlotWidget()
+        widget.on_toggle_code(on_toggle_callback)
+        widget.code_checkbox.value = True
+        assert count == 1
+
+
+class TestExporationBaseClass:
```

Comment:
oops as I was checking off FINC'd comments I see this is misspelled -  TestExporationBaseClass -> Test**Exploration**BaseClass

Reply:
fixed